### PR TITLE
[Function]support function bin on vectorized engine

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
cc: @HappenLee

File: gensrc/script/doris_builtins_functions.py [hunk 1]
```diff
@@ -610,7 +610,7 @@
             '15FunctionContextENS2_18FunctionStateScopeE', '', ''],
 
     [['bin'], 'VARCHAR', ['BIGINT'],
-            '_ZN5doris13MathFunctions3binEPN9doris_udf15FunctionContextERKNS1_9BigIntValE', '', '', '', ''],
+            '_ZN5doris13MathFunctions3binEPN9doris_udf15FunctionContextERKNS1_9BigIntValE', '', '', 'vec', ''],
```
Comment (HappenLee) [inline gensrc/script/doris_builtins_functions.py:613]: 应该标记为always nullable的

File: be/src/vec/functions/math.cpp [hunk 2]
```diff
@@ -313,6 +314,67 @@ struct NameDegrees {
...
+    static constexpr auto name = "bin";
+    static FunctionPtr create() { return std::make_shared<FunctionBin>(); }
+    String get_name() const override { return name; }
+    size_t get_number_of_arguments() const override { return 1; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return make_nullable(std::make_shared<DataTypeString>());
+    }
+
+    bool use_default_implementation_for_nulls() const override { return false; }
```
Comment (HappenLee) [inline be/src/vec/functions/math.cpp:328]: 这里设置为false了，但是代码段里没有处理null的对应逻辑

**Review by HappenLee — COMMENTED** [review]
有一些问题，并需要补充单测

File: be/src/vec/functions/math.cpp (hunk 2 above)
Comment (Gabriel39) [reply be/src/vec/functions/math.cpp]: done

File: gensrc/script/doris_builtins_functions.py (hunk 1 above)
Comment (HappenLee) [reply gensrc/script/doris_builtins_functions.py:613]: DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
        return make_nullable(std::make_shared<DataTypeString>());
    }
你在这里默认给它make nullable了，两个地方要匹配。否则你要报make_nullable的函数去掉

File: be/src/vec/functions/math.cpp [hunk 3]
```diff
@@ -313,6 +314,64 @@ struct NameDegrees {
...
+        for (int i = 0; i < size; ++i) {
+            StringOP::push_value_string(binImpl(data[i]), i, res_chars, res_offsets);
+        }
+
+        block.get_by_position(result).column =
+                ColumnNullable::create(std::move(res), std::move(null_map));
+    }
+
+private:
+    static std::string binImpl(Int64 value) {
```
Comment (HappenLee) [inline be/src/vec/functions/math.cpp:363]: doris之中统一使用下划线命名法，这里辛苦修改一下

Comment (Gabriel39) [reply be/src/vec/functions/math.cpp]: done

File: be/src/vec/functions/math.cpp [hunk 4]
```diff
@@ -313,6 +314,64 @@ struct NameDegrees {
...
+                        size_t input_rows_count) override {
+        bin_execute(block, arguments, result, input_rows_count);
+        return Status::OK();
+    }
+
+    static void bin_execute(Block& block, const ColumnNumbers& arguments, size_t result,
+                            size_t input_rows_count) {
+        DCHECK_EQ(arguments.size(), 1);
+
+        auto null_map = ColumnUInt8::create(input_rows_count, 0);
```
Comment (HappenLee) [inline be/src/vec/functions/math.cpp:340]: 这个函数的模式应该不是always nullable的模式，你这里没有额外处理null的逻辑。所以直接访问DataTypeString，不用加make_nullable

File: gensrc/script/doris_builtins_functions.py [hunk 5]
```diff
@@ -610,7 +610,7 @@
             '15FunctionContextENS2_18FunctionStateScopeE', '', ''],
 
     [['bin'], 'VARCHAR', ['BIGINT'],
-            '_ZN5doris13MathFunctions3binEPN9doris_udf15FunctionContextERKNS1_9BigIntValE', '', '', '', ''],
+            '_ZN5doris13MathFunctions3binEPN9doris_udf15FunctionContextERKNS1_9BigIntValE', '', '', 'vec', 'ALWAYS_NULLABLE'],
```
Comment (HappenLee) [inline gensrc/script/doris_builtins_functions.py:613]: 删除always_nullable的逻辑

**Pull request merged**